checker/src/main/java/org/checkerframework/checker/signedness/SignednessVisitor.java

```diff
@@ -47,10 +47,6 @@ private PrimitiveTypeTree primitiveTypeCast(Tree node) {
 
         TypeCastTree cast = (TypeCastTree) node;
         Tree castType = cast.getType();
-        // All types should be wrapped in some annotation
-        if (castType.getKind() != Kind.ANNOTATED_TYPE) {
```

Comment:
This is checking that castType is an AnnotatedTypeTree.   I've added a test case that crashes without this check.

Reply:
@smillst Thanks!  I think I should undo the code change and merge the additional test cases.
Is that correct?

Reply:
Yes.